### ISSUE TITLE
MP3 seeking will intermittently make it jump to the end and stop playing.

### ISSUE THREAD

**Issue by wolfbeast** [MoonchildProductions/Pale-Moon#841]
What the title says.
As part of my testing for #833 I noticed some odd behavior when seeking in the MP3 file. There would sometimes be a noticeable delay before playing resumes and other times playing would completely stop, with the cursor jumping to the very end of the stream.

This is also the case in 27.0.3-release, so not related to anything added in that commit, hence a separate issue.

**Comment by wolfbeast** [comment]
BMO references that may be relevant:
https://bugzilla.mozilla.org/show_bug.cgi?id=1089586
https://bugzilla.mozilla.org/show_bug.cgi?id=1208953
https://bugzilla.mozilla.org/show_bug.cgi?id=1159343
https://bugzilla.mozilla.org/show_bug.cgi?id=1163667

**Comment by wolfbeast** [comment]
This is also present in v27 a1, prior to most media changes. Likely this is a media bug in the underlying Mozilla code in the v38 platform.

**Comment by wolfbeast** [comment]
I can't seem to reproduce this with MP4 video, so assuming this is MP3 only.

**Comment by wolfbeast** [comment]
OK, this is not present in our latest common code point with Firefox platform code, so it must have been introduced early in Tycho development. That makes this a regression in our court.

**Comment by wolfbeast** [comment]
Made a mistake. a1 does not display this issue, public beta 1 does.

**Comment by wolfbeast** [comment]
Last good commit: eaa21516c7c71e24314fced1e834f5c41f6880b4 
First bad commit: 5167abf3947837e01586ae8a70e05f897a6b51a8 

Culprit: Making MDSM async in PR #540 -- @trav90 this is your work; did you make a mistake in porting that code or does it rely on other code changes... and TBH, why do we even need this to be async?

**Comment by trav90** [comment]
I think the culprit is commit a0125e4, which was landed by Mozilla (together with a few other fixes) to fix  regression caused by the MDSM changes in PR #540. However, it's apparently also talked about in a different BMO bug where it's mentioned as a "nice to have" instead of actually a fix to the problem. :roll_eyes: I think it should be safe to back out that commit. I'll run a build and confirm.

**Comment by trav90** [comment]
Should be resolved by PR #843.

**Comment by wolfbeast** [comment]
I built in parallel and can confirm.
